modules/data_warehouse/bigquery.tf

```diff
@@ -0,0 +1,334 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# Set up BigQuery resources
+# # Create the BigQuery dataset
+resource "google_bigquery_dataset" "ds_edw" {
+  project                    = module.project-services.project_id
+  dataset_id                 = "ds_edw"
+  friendly_name              = "My EDW Dataset"
+  description                = "My EDW Dataset with tables"
+  location                   = var.region
+  labels                     = var.labels
+  delete_contents_on_destroy = var.force_destroy
+}
+
+# # Create a BigQuery connection
+resource "google_bigquery_connection" "ds_connection" {
+  project       = module.project-services.project_id
+  connection_id = "ds_connection"
+  location      = var.region
+  friendly_name = "Storage Bucket Connection"
+  cloud_resource {}
+}
+
+# # Grant IAM access to the BigQuery Connection account for Cloud Storage
+resource "google_storage_bucket_iam_binding" "bq_connection_iam_object_viewer" {
+  bucket = google_storage_bucket.raw_bucket.name
+  role   = "roles/storage.objectViewer"
+  members = [
+    "serviceAccount:${google_bigquery_connection.ds_connection.cloud_resource[0].service_account_id}",
+  ]
+
+  depends_on = [
+    google_bigquery_connection.ds_connection,
+  ]
+}
+
+# # Create a BigQuery external table
+resource "google_bigquery_table" "tbl_edw_taxi" {
+  dataset_id          = google_bigquery_dataset.ds_edw.dataset_id
+  table_id            = "taxi_trips"
+  project             = module.project-services.project_id
+  deletion_protection = var.deletion_protection
+
+  external_data_configuration {
+    autodetect    = true
+    connection_id = "${module.project-services.project_id}.${var.region}.ds_connection"
+    source_format = "PARQUET"
+    source_uris   = ["gs://${google_storage_bucket.raw_bucket.name}/new-york-taxi-trips/tlc-yellow-trips-2022/taxi-*.Parquet"]
+
+  }
+
+  schema = <<EOF
+[
```

Comment:
It might be better to keep this as a standalone file and read it in TF.

Reply:
Done